tests/recognizer-tests.js

```diff
@@ -30,6 +30,16 @@ test("A unicode route recognizes", function() {
   equal(router.recognize("/uniçø∂∑"), null);
 });
 
+test("A unicode with special char route recognizes", function() {
```

Comment:
`encodeURIComponent("?#[]@!:$&'()*+,/=;")` returns 
"%3F%23%5B%5D%40!%3A%24%26'()*%2B%2C%2F%3D%3B"

so `! ' ( ) *` are characters that does not get encoded.